### DISCUSSION TITLE
Minix neo U9-H

### DISCUSSION THREAD

**Discussion by SadarSSI** [devmfc/debian-on-amlogic#1] [Q&A]
Hi, 

First thank you for your work !
I would like to understand how you manage to make a debian .img file with a vfat and dtb partition. Does detailed documentation exist somewhere?

Otherwise, I'm looking how to integrate the minix neo U9-H (The Minix Neo U9-H is an Android TV box that features an Amlogic S912-H processor)

Here is what I did:

I saw that there was a meson-gxm-minix-neo-u9h.dtb file in /dtb-6.1.1-meson64/amlogic/meson-gxm-minix-neo-u9h.dtb

I created a line in the boot.config file with
box=s905x_neo

I copied the /box-config/box-s905x_generic.config file to /box-config/box-s905x_neo.config

I modified the contents of the /box-config/box-s905x_neo.config file to load the meson-gxm-minix-neo-u9h.dtb file

It boots, but I don't have access to the console. What can I do because I am a beginner in this type of boot!

file 
>\# SPDX-License-Identifier: GPL-2.0+
> \# Version 2023 by Sadar
> \#
> \# Box: Minix Neo U9-H
> \#
> \# Comments are only allowed at the beginning of a line
> 
> dtb_img=dtb-${release}/amlogic/gxm_q200_2g_minix_neo_u9.dtb
> 
> load_config0=gxl-default-${variant}.config
> 
> \# warning, make sure this file ends with a newline!

What do you think ? 
Do i need to do anything else ?

Thanks for your advices

NB: my goal is to install debian minimal on the 16gb emmc. I don't want to run Kodi, just a OpenSSH server, DHCP server and AdguardHome, which is <5GB on a Rapsberry 3B+

**Comment by devmfc** [comment]
Hi,

Mainline linux has a dtb for this box I think (meson-gxm-minix-neo-u9h.dtb). So why would you try to use a dtb from coreelec? Which will probably not work by the way. Depending on which generation coreelec you used,  the kernel version of coreeleec is 4.9 or 5.4. The dtb's of these kernels are not compatible with mainline.

For gxm boxes I think you don't need the ugly hack I use for gxl (load_config0=gxl-default-${variant}.config), so I would try without this line first.

So I think the config file should be something like:
~~~
# SPDX-License-Identifier: GPL-2.0+
# Version 2023 by Sadar
#
# Box: Minix Neo U9-H
#
# Comments are only allowed at the beginning of a line

dtb_img=dtb-${release}/amlogic/meson-gxm-minix-neo-u9h.dtb

rd_img=
dtb_addr=0x4000000

# warning, make sure this file ends with a newline!
~~~
Do you use a usb flash drive or sdcard? For Amlogic boxes I would recommend to use a USB flash drive.

What do you mean with _how you manage to make a debian .img file with a vfat and dtb partition_?

**Comment by SadarSSI** [comment]
Thank you for your prompt response

Indeed the coreelec dtd file does not work but the one present by default works.

It boots but stays on: "random crng init done"

I'll try to make the changes you told me.

I boot from an sdcard, it doesn't seem to work on a usb key, but I will test again later

**Reply by devmfc** [reply]
if you boot from sdcard remove the line in boot.config that says _debug=y_ and uncomment the line with _root=/dev/mmcblk0p2_

**Reply by SadarSSI** [reply]
Waaaaaaa!!! Awesome
I just have to install the system on the emmc memory, I believe that debian allows this

I will post the file /box-config/box-s905x_neo.config which works like a charm

Do you know where to find the documentation to recreate this configuration with the boot part in vfat and the linux partition in ext4?

Thanks for your help

**Reply by devmfc** [reply]
👍 Great!

By the way, your box is a S912 (octa core) not a S905X (quad core).

But I still do not understand the question _Do you know where to find the documentation to recreate this configuration with the boot part in vfat and the linux partition in ext4_? You can make any partition configuration you like with just the _parted_ tool and then copy all the files from your bootstrapped Debian to it?

Something like here: [https://a-delacruz.github.io/debian/debian-arm64.html](https://a-delacruz.github.io/debian/debian-arm64.html)

**Comment by SadarSSI** [comment]
Yes, I know how to make a custom iso but I haven't had the opportunity to use bootstrap yet. 

Yes, as much as I know how to make a custom iso but I haven't had the opportunity to use bootstrap yet. I think this is the part I was missing because for the rest: dd, parted etc I do how.

All I have to do is transfer the system to the emmc because otherwise the box will reboot on android ;-((

For the S912 proc, you are right, I should rename the file to /box-config/box-s912_neo.config


Thank you for your advice and your help, because I would not have known how to put the right parameters in the config files

All I have to do is transfer the system to the emmc because otherwise the box will reboot on android ;-((

For the S912 proc, you are right, I should rename the file to /box-config/box-s912_neo.config

Thank you for your advice and your help, because I would not have known how to put the right parameters in the config files
